### ISSUE TITLE
Formalize code version in online doc URLs and shorten them?

### ISSUE THREAD

**Issue by giacomofiorin** [Colvars/colvars#605]
The HTML and PDF doc links for the `master` branch currently look like this:
```
https://colvars.github.io/colvars-refman-lammps/colvars-refman-lammps.html
https://colvars.github.io/pdf/colvars-refman-lammps.pdf
```

The corresponding links for specific releases are similar to those above, except for being nested inside a sub-folder named `<backend>-<version>`, e.g. the following for NAMD:
```
https://colvars.github.io/namd-2.14/colvars-refman-namd/colvars-refman-namd.html
https://colvars.github.io/namd-2.14/pdf/colvars-refman-namd.pdf
```

What I propose is the following:
1. Do a similar treatment for `master`, as well as any other branch where a doc update might be useful.
2. Shorten the URLs by putting all relevant files directly in that folder.

In summary, I'd like to adopt the following format for the online docs:
```
https://colvars.github.io/master/colvars-refman-namd.html
https://colvars.github.io/master/colvars-refman-namd.pdf
```

For backward compatibility, we can just keep updating the current HTML and PDF docs, since they are already being linked to in public forums.

Any thoughts?

**Comment by jhenin** [comment]
This sounds very good to me.
